test/mock_error.hpp

```diff
@@ -10,12 +10,12 @@
 #define MOCK_TEST_MOCK_ERROR_HPP_INCLUDED
 
 #define MOCK_ERROR_POLICY mock_error
+#include <turtle/detail/singleton.hpp>
 #include <boost/lexical_cast.hpp>
 #include <boost/test/unit_test.hpp>
-#include <boost/test/utils/trivial_singleton.hpp>
 #include <stdexcept>
 
-struct mock_error_data_t : boost::unit_test::singleton< mock_error_data_t >
+struct mock_error_data_t : turtle::detail::singleton< mock_error_data_t >
```

Comment:
g++ doesn't like this it seems https://travis-ci.org/mat007/turtle/jobs/455635618
```
In file included from test_sequence.cpp:9:0:
mock_error.hpp:18:28: error: ‘turtle’ has not been declared
 struct mock_error_data_t : turtle::detail::singleton< mock_error_data_t >
```

Reply:
Ah yes, why is the namespace of `turtle/detail/foo.hpp` `mock::detail::foo`? Shouldn't the toplevel namespace be `turtle`?

Changed it now to `mock`